src/posthog-surveys.ts

```diff
@@ -0,0 +1,75 @@
+import { PostHog } from './posthog-core'
+import { SURVEYS } from './posthog-persistence'
+import { SurveyCallback } from 'types'
+
+export class PostHogSurveys {
+    instance: PostHog
+
+    constructor(instance: PostHog) {
+        this.instance = instance
+    }
+
+    getSurveys(callback: SurveyCallback, forceReload = false) {
+        const existingSurveys = this.instance.get_property(SURVEYS)
+        if (!existingSurveys || forceReload) {
+            this.instance._send_request(
+                `${this.instance.get_config('api_host')}/api/surveys/?token=${this.instance.get_config('token')}`,
+                {},
+                { method: 'GET' },
+                (response) => {
+                    const surveys = response.surveys
+                    this.instance.persistence.register({ [SURVEYS]: surveys })
+                    return callback(surveys)
+                }
+            )
+        } else {
+            return callback(existingSurveys)
+        }
+    }
+
+    getActiveMatchingSurveys(callback: SurveyCallback, forceReload = false) {
+        this.getSurveys((surveys) => {
+            const activeSurveys = surveys.filter((survey) => {
+                return !!(survey.start_date && !survey.end_date)
```

Comment:
we ought to check this date in relation to `now` as well, i.e. start < now < end. I'm pretty sure we'd want surveys to be schedulable, which is where this check comes in handy

Reply:
yeah though we don't need it now until we have scheduling and such, which will require an update to the logic and tests then, wyt?

Reply:
yep all good